pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -869,10 +890,18 @@ func (c *Reconciler) createTaskRun(ctx context.Context, taskRunName string, para
 
 	var pipelinePVCWorkspaceName string
 	var err error
+	// Propagate workspaces from pipelineRun to the taskRun.
 	tr.Spec.Workspaces, pipelinePVCWorkspaceName, err = c.getTaskrunWorkspaces(ctx, pr, rpt)
 	if err != nil {
 		return nil, err
 	}
+	// By this time, params should be propagated so we can
+	// validate their usage. However, we propagate the workspaces from pipelineRun directly to
+	// the pipeline tasks so we cannot validate workspaces. TaskRun validation will ensure that
```

Comment:
should we make workspace propagation work in the same way as param propagation? would it make sense to create an issue to track validation of propagated workspaces?

Reply:
I was thinking about that as well. We don't gain anything functionally but it does make it consistent with parameter propagation and improved readability. But I was wondering if I should focus on validating workspaces for referenced pipelines in this PR and a new one that syncs implementation of propagated params and workspaces.

Reply:
I think that would be reasonable to focus this PR on validating remote pipelines if you create a new issue to track the remaining work!

Reply:
Done! https://github.com/tektoncd/pipeline/issues/6847